cmd/juju/application/store/store.go

```diff
@@ -27,25 +23,3 @@ func AddCharmFromURL(client CharmAdder, curl *charm.URL, origin commoncharm.Orig
 	}
 	return curl, resultOrigin, nil
 }
-
-// NewCharmStoreClient is called to obtain a charm store client.
-// It is defined as a variable so it can be changed for testing purposes.
-var NewCharmStoreClient = func(client *httpbakery.Client, csURL string) *csclient.Client {
-	return csclient.New(csclient.Params{
-		URL:            csURL,
-		BakeryClient:   client,
-		UserAgentValue: version.UserAgentVersion,
-	})
-}
-
-// NewCharmStoreAdaptor combines charm store functionality with the ability to get a macaroon.
-func NewCharmStoreAdaptor(client *httpbakery.Client, csURL string) *CharmStoreAdaptor {
-	cstoreClient := NewCharmStoreClient(client, csURL)
-	return &CharmStoreAdaptor{
-		CharmrepoForDeploy: charmrepo.NewCharmStoreFromClient(cstoreClient),
-	}
-}
-
-type CharmStoreAdaptor struct {
-	CharmrepoForDeploy
```

Comment:
CharmrepoForDeploy can be removed from the store/interface.go file now. This will also allow for `func (f *fakeDeployAPI) ResolveWithPreferredChannel` to be removed from application/deploy_test.go